pincer/cog.py

```diff
@@ -0,0 +1,113 @@
+# Copyright Pincer 2021-Present
+# Full MIT License can be found in `LICENSE` at the project root.
+
+from __future__ import annotations
+from asyncio import ensure_future
+
+from importlib import reload, import_module
+from types import ModuleType
+from typing import TYPE_CHECKING
+
+from . import client as _client
+from .commands.chat_command_handler import ChatCommandHandler
+from .commands.interactable import Interactable
+from .exceptions import CogAlreadyExists, CogNotFound
+
+if TYPE_CHECKING:
+    from typing import Type
+    from .client import Client
+
+
+def get_cog_name(cog: Type[Cog]):
+    """Gets the path to import a cog. This is used as a unique identifier"""
+    try:
+        return f"{cog.__module__}.{cog.__name__}"
+    except AttributeError:
+        return f"{cog.__module__}.{cog.__class__.__name__}"
+
+
+def load_cog(client: Client, cog: Type[Cog]):
+    """Loads a cog
+
+    Parameters
+    ----------
+    cog : Type[:class:`~pincer.cog.Cog`]
+        The cog to load.
+    """
+    if cog in ChatCommandHandler.managers:
+        raise CogAlreadyExists(
+            f"Cog `{cog}` is trying to be loaded but already exists."
+        )
+
+    cog_manager = cog(client)
+
+    ChatCommandHandler.managers.append(cog_manager)
```

Comment:
Since cogs are inheriting our `Cog`, cannot we use `__init__subclass` to append the manager?

Reply:
I don't think so. If you have multiple cogs in the same file I think it would break when you reloaded one of them.